src/components/dashboard/main-content/course-enrollments/data/hooks.js

```diff
@@ -125,114 +130,132 @@ export const useCourseEnrollments = ({
  * @param {String} args.mode The mode of the course. Used as a gating mechanism for upgradability
  * default: false
  * @returns {Object} {
- *     licenseUpgradeUrl: undefined,
- *     couponUpgradeUrl: undefined,
- *     learnerCreditUpgradeUrl: undefined,
  *     subsidyForCourse: undefined,
  *     courseRunPrice: undefined,
- *     }
+ *     hasUpgradeAndConfirm: false,
+ * }
  */
 export const useCourseUpgradeData = ({
   courseRunKey,
   mode,
 }) => {
   const location = useLocation();
-  // We determine whether the course mode is such that it can be upgraded
+  // Determine whether the course mode is such that it can be upgraded
   const canUpgradeToVerifiedEnrollment = [COURSE_MODES_MAP.AUDIT, COURSE_MODES_MAP.HONOR].includes(mode);
   const { authenticatedUser } = useContext(AppContext);
   const { data: enterpriseCustomer } = useEnterpriseCustomer();
-  const { data: customerContainsContent } = useEnterpriseCustomerContainsContent([courseRunKey]);
+  const { data: customerContainsContent } = useEnterpriseCustomerContainsContent([courseRunKey], {
+    enabled: canUpgradeToVerifiedEnrollment,
+  });
 
-  // TODO: Remove authenticatedUser?.administrator flag when rolling out ENT-9135
+  // TODO: Remove `isLearnerCreditUpgradeEnabled` flag when rolling out ENT-9135
   // Metadata required to allow upgrade via applicable learner credit
-  const { data: learnerCreditMetadata } = useCanUpgradeWithLearnerCredit(
-    [courseRunKey],
-    { enabled: authenticatedUser?.administrator && canUpgradeToVerifiedEnrollment },
-  );
+  const isLearnerCreditUpgradeEnabled = authenticatedUser.administrator || hasFeatureFlagEnabled('LEARNER_CREDIT_AUDIT_UPGRADE');
+  const { data: learnerCreditMetadata } = useCanUpgradeWithLearnerCredit(courseRunKey, {
+    enabled: isLearnerCreditUpgradeEnabled && canUpgradeToVerifiedEnrollment,
+  });
```

Comment:
[inform] Feature flagged to only call the `can-redeem` API for staff users or when a `?feature=LEARNER_CREDIT_AUDIT_UPGRADE` query parameter is present in the URL.